tests/models/bert/test_modeling_bert.py

```diff
@@ -584,6 +584,9 @@ def test_for_warning_if_padding_and_no_attention_mask(self):
 
         # Check for warnings if the attention_mask is missing.
         logger = logging.get_logger("transformers.modeling_utils")
+        # clear cache so we can test the warning is emitted (from `warning_once`).
```

Comment:
`warn_if_padding_and_no_attention_mask` uses `warning_once`, so we need to clear cache to make sure we will get the expected warning.